### PR TITLE
plot_singles_timefreq: plot veto segments, fix gate plotting and caption

### PR DESCRIPTION
**Pull request by titodalcanton**
Example plot with veto segments: https://www.atlas.aei.uni-hannover.de/~tito/LSC/o2/H1-PLOT_SINGLES_TIMEFREQ_2-1184112018-1105200.png

### CODE REVIEW

**Comment by tdent** [comment]
I'm on holiday / travel until Monday afternoon, so assign to someone else if you want it quickly.

File: bin/hdfcoinc/pycbc_plot_singles_timefreq [hunk 1]
```diff
@@ -200,13 +200,22 @@ for gate_type, hatch_style in [('file', '\\'), ('auto', '/')]:
...
             continue
         ax.axvspan(gt - gw - center_time, gt + gw - center_time,
                    hatch=hatch_style, facecolor='none', edgecolor='#00ff00')
 
+if opts.veto_file:
+    logging.info('Loading and plotting veto segments')
+    veto_segs = pycbc.events.veto.select_segments_by_definer(
+            opts.veto_file, ifo=opts.detector)
+    for seg in veto_segs:
+        ax.axvspan(seg[0] - center_time, seg[1] - center_time,
```
Comment (tdent) [inline bin/hdfcoinc/pycbc_plot_singles_timefreq]: probably we would rather not plot every segment in a whole run, there could be >thousands. prefer to do similar to the gates and preselect the ones that would appear on the plot.

**Comment by duncan-brown** [comment]
@titodalcanton this branch will need to be rebased to master in order for Travis to complete. See https://github.com/ligo-cbc/pycbc/issues/1758

**Comment by titodalcanton** [comment]
@tdent I addressed your comment.

**Pull request merged**